src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleObject.cs

```diff
@@ -47,6 +71,23 @@ private static bool HandleEndObject(JsonSerializerOptions options, ref ReadStack
 
             state.Current.JsonClassInfo.UpdateSortedPropertyCache(ref state.Current);
 
+            if (state.Current.IsDictionary())
+            {
+                ReadStackFrame frame = state.Current;
+
+                int keyCount = ((IList)frame.TempDictKeys).Count;
+
+                Debug.Assert(keyCount == (frame.TempDictValues).Count);
```

Comment:
nit: Remove extra brackets around `frame.TempDictValues` or add a missing cast to `IList`.